### PR TITLE
[ZEPPELIN-2572] multiple paragraphs actions

### PR DESCRIPTION
**Pull request by Savalek**
### What is this PR for?

This PR adds the ability to perform actions on several paragraphs simultaneously. He adds the checkbox to each paragraph. When you select at least one paragraph in the action bar, buttons appear that perform actions on the selected paragraphs.

### What type of PR is it?
Feature

### What is the Jira issue?
[ZEPPELIN-2572](https://issues.apache.org/jira/browse/ZEPPELIN-2572)

### Screenshots (if appropriate)
mult_v2

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

File: zeppelin-web/src/app/notebook/notebook.controller.js [hunk 1]
```diff
@@ -606,6 +609,120 @@ function NotebookCtrl($scope, $route, $routeParams, $location, $rootScope,
...
+  };
+
+  $scope.clearSelection = function() {
+    if ($scope.selectedParagraphsIds !== null) {
+      $scope.selectedParagraphsIds.clear();
+    }
+  };
+
+  $scope.isSelectionMode = function() {
+    if ($scope.selectedParagraphsIds === null || $scope.selectedParagraphsIds.size === 0) {
```
Comment (Tagar) [inline zeppelin-web/src/app/notebook/notebook.controller.js]: What do you think if multi-paragraph toolbar will popup when at least two paragraphs selected?

File: zeppelin-web/src/app/notebook/notebook.controller.js [hunk 2]
```diff
@@ -606,6 +609,120 @@ function NotebookCtrl($scope, $route, $routeParams, $location, $rootScope,
...
+        }
+      },
+    });
+  };
+
+  $scope.selectedParagraphsAction = function(type, requestConfirm) {
+    let noteId = $scope.note.id;
+    let action;
+    let broadcastAction = function(data) {
+      return () => $scope.$broadcast('multiplyAction', type, $scope.selectedParagraphsIds, data);
```
Comment (Tagar) [inline zeppelin-web/src/app/notebook/notebook.controller.js]: replace "multiplyAction" with "multipleAction" in all places

Comment (Savalek) [reply zeppelin-web/src/app/notebook/notebook.controller.js]: Thanks Tagar. I will fix it.

File: zeppelin-web/src/app/notebook/notebook.controller.js (hunk 1 above)
Comment (Savalek) [reply zeppelin-web/src/app/notebook/notebook.controller.js]: @Tagar, I do not quite understand what you mean.

**Comment by Savalek** [comment]
Now checkBox to select will be displayed if: 
1. Mouse located in paragraph
2. At least one paragraph selected
mult_v2_change_css

File: zeppelin-web/src/app/notebook/notebook.controller.js (hunk 1 above)
Comment (Tagar) [reply zeppelin-web/src/app/notebook/notebook.controller.js]: consider changing this line to 
`if ($scope.selectedParagraphsIds === null || $scope.selectedParagraphsIds.size < 2) {` 
so the toolbar for multi-paragraph actions will show up when at least 2 paragraphs are selected.

**Comment by Tagar** [comment]
@Leemoonsoo please help reviewing this 

I personally think that color palette of selected paragraphs can be better something else, but functionally it's a great improvement to core Zeppelin UI. 

Thanks.

**Pull request closed without merging**